chainstate/src/detail/transaction_verifier/mod.rs

```diff
@@ -181,97 +185,27 @@ impl<'a, S: BlockchainStorageRead> TransactionVerifier<'a, S> {
-                    .ok_or(ConnectTransactionError::MissingOutputOrSpent)?;
-                entry.insert(CachedInputsOperation::Read(tx_index))
-            }
-        };
-        Ok(())
-    }
-
-    fn get_output_value(
-        outputs: &[TxOutput],
-        output_index: usize,
-        spender_id: Spender,
-    ) -> Result<&OutputValue, ConnectTransactionError> {
-        let output =
-            outputs
-                .get(output_index)
-                .ok_or(ConnectTransactionError::OutputIndexOutOfRange {
-                    tx_id: Some(spender_id),
-                    source_output_index: output_index,
-                })?;
-        Ok(output.value())
-    }
-
-    fn calculate_coins_total_inputs(
-        &self,
-        inputs: &[TxInput],
-    ) -> Result<Amount, ConnectTransactionError> {
-        let mut total = Amount::from_atoms(0);
-        for (_input_idx, input) in inputs.iter().enumerate() {
-            let outpoint = input.outpoint();
-            let tx_index = match self.tx_index_cache.get(&outpoint.tx_id()) {
-                Some(tx_index_op) => match tx_index_op {
-                    CachedInputsOperation::Write(tx_index) => tx_index,
-                    CachedInputsOperation::Read(tx_index) => tx_index,
-                    CachedInputsOperation::Erase => {
-                        return Err(ConnectTransactionError::PreviouslyCachedInputWasErased)
-                    }
-                },
-                None => return Err(ConnectTransactionError::PreviouslyCachedInputNotFound),
-            };
-            let output_index = outpoint.output_index() as usize;
-
-            let output_amount = match tx_index.position() {
-                common::chain::SpendablePosition::Transaction(tx_pos) => {
-                    let tx = self
-                        .db_tx
-                        .get_mainchain_tx_by_position(tx_pos)
-                        .map_err(ConnectTransactionError::from)?
-                        .ok_or_else(|| {
-                            ConnectTransactionError::InvariantErrorTransactionCouldNotBeLoaded(
-                                tx_pos.clone(),
-                            )
-                        })?;
-
-                    Self::get_output_value(tx.outputs(), output_index, tx.get_id().into())
-                        .cloned()?
-                }
-                common::chain::SpendablePosition::BlockReward(block_id) => {
-                    let outputs = self.block_reward_outputs(block_id)?;
-                    Self::get_output_value(&outputs, output_index, (*block_id).into()).cloned()?
-                }
-            };
-            match output_amount {
-                OutputValue::Coin(output_amount) => {
-                    total = (total + output_amount)
-                        .ok_or(ConnectTransactionError::InputAdditionError)?
-                }
-            }
+        if let Entry::Vacant(entry) = self.tx_index_cache.entry(outpoint.tx_id()) {
```

Comment:
I have this deep grudge against if let because it's heavily overused. I use it only in very specific situations. I prefer match because it lays out the possibilities clearly for the developer, and it doesn't allow you to miss new arms added in the future (irrelevant for this case). There's no practical benefit of the change you did here. If you want to get rid of the reference returned, you can return () in the other arm, but I still find this unnecessary as it'll be optimized out by the compiler.

Reply:
The reason I changed match to if let here is that it is more clean. I'm new to the code and things like unused variables and empty branches (that returns useless refs) are very distractive.

Reply:
> The reason I changed match to if let here is that it is more clean

That's quite subjective. To me, I call match cleaner because it's easier to maintain and clearer to the developer what the options are.

Please restore it. I understand it's not ideal for your taste, but please bear with me in such issues as I have goals that pertain to security more than things being nicer/uglier.

Reply:
ok